packages/SwingSet/src/kernel/kernel.js

```diff
@@ -445,7 +445,7 @@ export default function buildKernel(
       const vatID = kernelKeeper.ownerOfKernelObject(target);
       if (vatID) {
         await deliverToVat(vatID, target, msg);
-      } else {
+      } else if (msg.result) {
```

Comment:
I saw one more that needs to be guarded: line 413, in `deliverToVat`, just after the `vatWarehouse.lookup`. Please add an inner `if (msg.result)` around that:
```js
if (!vatWarehouse.lookup(vatID)) {
  if (msg.result) {
    resolveToError(msg.result, VAT_TERMINATION_ERROR);
  }
} else {
 ... deliverAndLogToVat clause
```

Reply:
Done.